sql/analyzer/process_truncate.go

```diff
@@ -0,0 +1,156 @@
+package analyzer
+
+import (
+	"strings"
+
+	"github.com/dolthub/vitess/go/vt/sqlparser"
+
+	"github.com/dolthub/go-mysql-server/sql"
+	"github.com/dolthub/go-mysql-server/sql/plan"
+)
+
+func processTruncate(ctx *sql.Context, a *Analyzer, n sql.Node, scope *Scope) (sql.Node, error) {
+	span, _ := ctx.Span("processTruncate")
+	defer span.Finish()
+
+	deletePlan, ok := n.(*plan.DeleteFrom)
+	if ok {
+		return deleteToTruncate(ctx, a, deletePlan)
+	}
+	truncatePlan, ok := n.(*plan.Truncate)
+	if ok {
+		_, err := validateTruncate(ctx, a, truncatePlan.DatabaseName(), truncatePlan.Child())
+		if err != nil {
+			return nil, err
+		}
+		return truncatePlan, nil
+	}
+	return n, nil
+}
+
+func deleteToTruncate(ctx *sql.Context, a *Analyzer, deletePlan *plan.DeleteFrom) (sql.Node, error) {
+	tbl, ok := deletePlan.Child.(*plan.ResolvedTable)
+	if !ok {
+		return deletePlan, nil
+	}
+	tblName := strings.ToLower(tbl.Name())
+
+	// auto_increment behaves differently for TRUNCATE and DELETE
+	for _, col := range tbl.Schema() {
+		if col.AutoIncrement {
+			return deletePlan, nil
+		}
+	}
+
+	dbName := ""
+	tblFound := false
+	//TODO: if multiple dbs have a table with the same name it's ambiguous, find a way to distinguish (will remove this pass)
+	for _, db := range a.Catalog.AllDatabases() {
+		dbTblNames, err := db.GetTableNames(ctx)
```

Comment:
I think instead of this, just use ctx.CurrentDatabase() and assert that it has a table with the name you got from the ResolvedTable node, erroring out if not found. This matches what we do in other places where we don't have a database qualified table but should.

Also use db.GetTableInsensitive